File: Library.Core.Web/src/app/view-models/Book/putBookView.ts
export class PutBookView {
  constructor(
    public id: number = 0,
    public name?: string,
    public yearOfPublishing: number = 0,
    public authors?: AuthorPutBookViewItem[],
    public publicationHouses?: PublicationHousePutBookViewItem[]
  ) { }
}

export class AuthorPutBookViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public birthday: number = 0,
    public deathday: number = 0
  ) { }
}

export class PublicationHousePutBookViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public adress?: string
  ) { }
}

File: Library.Core.ViewModelLayer/ViewModels/BookViewModels/PutBookView.cs
using System.Collections.Generic;

namespace Library.Core.ViewModelLayer.ViewModels.Book
{
    public class PutBookView
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int YearOfPublishing { get; set; }

        public ICollection<AuthorPutBookViewItem> Authors { get; set; }
        public ICollection<PublicationHousePutBookViewItem> PublicationHouses { get; set; }

        public PutBookView()
        {
            Authors = new List<AuthorPutBookViewItem>();
            PublicationHouses = new List<PublicationHousePutBookViewItem>();
        }
    }

    public class AuthorPutBookViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Birthday { get; set; }
        public int Deathday { get; set; }

        public AuthorPutBookViewItem()
        {

        }
    }

    public class PublicationHousePutBookViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Adress { get; set; }

        public PublicationHousePutBookViewItem()
        {

        }
    }
}
